The cancelled test:

```
2021-02-12 23:39:17,250 ERROR [galaxy.jobs.handler] Exception in monitor_step
Traceback (most recent call last):
  File "/home/runner/.planemo/gx_venv_3.7/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 1277, in _execute_context
    cursor, statement, parameters, context
  File "/home/runner/.planemo/gx_venv_3.7/lib/python3.7/site-packages/sqlalchemy/engine/default.py", line 609, in do_execute
    cursor.execute(statement, parameters)
sqlite3.OperationalError: near "(": syntax error

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/runner/work/galaxy/galaxy/galaxy root/lib/galaxy/jobs/handler.py", line 281, in __monitor
    self.__monitor_step()
  File "/home/runner/work/galaxy/galaxy/galaxy root/lib/galaxy/jobs/handler.py", line 299, in __monitor_step
    self.__handle_waiting_jobs()
  File "/home/runner/work/galaxy/galaxy/galaxy root/lib/galaxy/jobs/handler.py", line 344, in __handle_waiting_jobs
    .filter(ranked.c.rank <= self.app.job_config.handler_ready_window_size).all()
  File "/home/runner/.planemo/gx_venv_3.7/lib/python3.7/site-packages/sqlalchemy/orm/query.py", line 3373, in all
    return list(self)
  File "/home/runner/.planemo/gx_venv_3.7/lib/python3.7/site-packages/sqlalchemy/orm/query.py", line 3535, in __iter__
    return self._execute_and_instances(context)
  File "/home/runner/.planemo/gx_venv_3.7/lib/python3.7/site-packages/sqlalchemy/orm/query.py", line 3560, in _execute_and_instances
    result = conn.execute(querycontext.statement, self._params)
  File "/home/runner/.planemo/gx_venv_3.7/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 1011, in execute
    return meth(self, multiparams, params)
  File "/home/runner/.planemo/gx_venv_3.7/lib/python3.7/site-packages/sqlalchemy/sql/elements.py", line 298, in _execute_on_connection
    return connection._execute_clauseelement(self, multiparams, params)
  File "/home/runner/.planemo/gx_venv_3.7/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 1130, in _execute_clauseelement
    distilled_params,
  File "/home/runner/.planemo/gx_venv_3.7/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 1317, in _execute_context
    e, statement, parameters, cursor, context
  File "/home/runner/.planemo/gx_
```

Seems plausibly related - seems like converter tests may be the only tests that extensively use sqlite?

--- a/lib/galaxy/jobs/__init__.py
+++ b/lib/galaxy/jobs/__init__.py
@@ -304,7 +304,7 @@ class JobConfiguration(ConfiguresHandlers):
         self.handler_assignment_methods = None
         self.handler_assignment_methods_configured = False
         self.handler_max_grab = None
-        self.handler_ready_job_window = None
+        self.handler_ready_window_size = None
         self.destinations = {}
         self.destination_tags = {}
         self.default_destination_id = None

--- a/lib/galaxy/jobs/handler.py
+++ b/lib/galaxy/jobs/handler.py
@@ -335,13 +335,21 @@ class JobHandlerQueue(Monitors):
             if self.app.config.user_activation_on:
                 job_filter_conditions = job_filter_conditions + (
                     or_((model.Job.user_id == null()), (model.User.active == true())),)
-            ranked = self.sa_session.query(model.Job, rank).enable_eagerloads(False) \
+            if self.sa_session.bind.name == 'sqlite':
+                query_objects = (model.Job,)
+            else:
+                query_objects = (model.Job, rank)
+            ready_query = self.sa_session.query(*query_objects).enable_eagerloads(False) \
                 .outerjoin(model.User) \
                 .filter(and_(*job_filter_conditions)) \
-                .order_by(model.Job.id).subquery()
-            jobs_to_check = self.sa_session.query(model.Job) \
-                .join(ranked, model.Job.id == ranked.c.id) \
-                .filter(ranked.c.rank <= self.app.job_config.handler_ready_window_size).all()
+                .order_by(model.Job.id)
+            if self.sa_session.bind.name == 'sqlite':
+                jobs_to_check = ready_query.all()
+            else:
+                ranked = ready_query.subquery()
+                jobs_to_check = self.sa_session.query(model.Job) \
+                    .join(ranked, model.Job.id == ranked.c.id) \
+                    .filter(ranked.c.rank <= self.app.job_config.handler_ready_window_size).all()
             # Filter jobs with invalid input states
             jobs_to_check = self.__filter_jobs_with_invalid_input_states(jobs_to_check)
             # Fetch all "resubmit" jobs

--- a/lib/galaxy/jobs/handler.py
+++ b/lib/galaxy/jobs/handler.py
@@ -339,7 +339,9 @@ class JobHandlerQueue(Monitors):
                 .outerjoin(model.User) \
                 .filter(and_(*job_filter_conditions)) \
                 .order_by(model.Job.id).subquery()
-            jobs_to_check = self.sa_session.query(ranked).filter(ranked.c.rank <= 1).all()
+            jobs_to_check = self.sa_session.query(model.Job) \
+                .join(ranked, model.Job.id == ranked.c.id) \
+                .filter(ranked.c.rank <= 1).all()
             # Filter jobs with invalid input states
             jobs_to_check = self.__filter_jobs_with_invalid_input_states(jobs_to_check)
             # Fetch all "resubmit" jobs

